File: src/dataset/Java/resistor-color-duo_exercism-johannesrave-education/gemini/GeminiFlash_ResistorColorDuo_v1.java
import java.util.List;
import java.util.Map;
import java.util.function.Function;
import java.util.stream.Collectors;

class ResistorColorDuo {
    private static final Map<String, Integer> colorCode = List.of(
            "black", "brown", "red", "orange", "yellow",
            "green", "blue", "violet", "grey", "white")
            .stream()
            .collect(Collectors.toMap(Function.identity(), List.of(
                    "black", "brown", "red", "orange", "yellow",
                    "green", "blue", "violet", "grey", "white")::indexOf));

    int value(String[] colors) {
        return colorCode.get(colors[0]) * 10 + colorCode.get(colors[1]);
    }
}

File: src/dataset/c/resistor-color-duo_exercism-viettronics868/gemini/GeminiFlash_resistor_color_duo_v1.c
#include "resistor_color_duo.h"
#include <stdint.h>

uint16_t color_code(const resistor_band_t *resis_band) {
    uint16_t value = 0;

    switch (resis_band[0]) {
        case BLACK:   value = 0; break;
        case BROWN:   value = 10; break;
        case RED:     value = 20; break;
        case ORANGE:  value = 30; break;
        case YELLOW:  value = 40; break;
        case GREEN:   value = 50; break;
        case BLUE:    value = 60; break;
        case VIOLET:  value = 70; break;
        case GREY:    value = 80; break;
        case WHITE:   value = 90; break;
        default:      value = 0; break; // Handle unexpected input
    }

    switch (resis_band[1]) {
        case BLACK:   value += 0; break;
        case BROWN:   value += 1; break;
        case RED:     value += 2; break;
        case ORANGE:  value += 3; break;
        case YELLOW:  value += 4; break;
        case GREEN:   value += 5; break;
        case BLUE:    value += 6; break;
        case VIOLET:  value += 7; break;
        case GREY:    value += 8; break;
        case WHITE:   value += 9; break;
        default:      break; // Handle unexpected input
    }

    return value;
}